__tests__/create-token.test.js

```diff
@@ -0,0 +1,71 @@
+import TestUtils from "./test-utils";
+
+describe("create-token api", () => {
+  it("should return 200 with a valid body", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/create-token")
+      .send({
+        name: "stub_token",
+        symbol: "03",
+        amount: 1,
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeDefined();
+  });
+
+  it("should create a token with amount as string", async () => {
+    const response = await TestUtils.request
+      .post("/wallet/create-token")
+      .send({
+        name: "stub_token",
+        symbol: "03",
+        amount: "1",
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    expect(response.status).toBe(200);
+    expect(response.body.hash).toBeDefined();
+  });
+
+  it("should not create a token without the required parameters", async () => {
+    ["name", "symbol", "amount"].forEach(async (field) => {
+      const token = {
+        name: "stub_token",
+        symbol: "03",
+        amount: 1,
+      };
+      delete token[field];
+      const response = await TestUtils.request
+        .post("/wallet/create-token")
+        .send(token)
+        .set({ "x-wallet-id": TestUtils.walletId });
+      expect(response.status).toBe(400);
+      expect(response.body.success).toBeFalsy();
+    });
+  });
+
+  it("should receive an error when trying to do concurrent create-token (lock/unlock behavior)", async () => {
+    const promise1 = TestUtils.request
+      .post("/wallet/create-token")
+      .send({
+        name: "stub_token",
+        symbol: "03",
+        amount: 1,
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+    const promise2 = TestUtils.request
+      .post("/wallet/create-token")
+      .send({
+        name: "stub_token",
+        symbol: "03",
+        amount: 1,
+      })
+      .set({ "x-wallet-id": TestUtils.walletId });
+
+    const [response1, response2] = await Promise.all([promise1, promise2]);
```

Comment:
I feel we should add tests for a high amount such as there's no HTR funds to do it, and also to try to create a token with the symbol HTR or the name Hathor.

Reply:
1. `tests for a high amount such as there's no HTR funds to do it`
Done
2. `try to create a token with the symbol HTR or the name Hathor`
It seems that this validation is done by the `fullnode`, so I can only mock this behavior. But mocking this `fullnode` error to assert it is retransmitted to `hathor-wallet-headless` does not sound like a valid effort.